src/util.rs

```diff
@@ -1,47 +1,191 @@
         for _ in 0..8 {
-            value = if (value & 1) == 1 {
-                (value >> 1) ^ poly
+            if (value & top_bit) != 0 {
+                value = (value << 1) ^ poly
             } else {
-                value >> 1
+                value <<= 1
             }
         }
-        table[i] = value;
+
+        if true == rfl {
+            value = reflect_short(value);
+        }
+
+        table[i as usize] = value;
     }
     table
 }
 
-pub fn make_table_crc32(poly: u32) -> [u32; 256] {
+/// Builds a CRC32 table using the standard CRC method
+/// If reflect==true, flip the individual byte bitwise, then flip the 32bit table value bitwise
+pub fn make_table_crc32(poly: u32, rfl: bool) -> [u32; 256] {
     let mut table = [0u32; 256];
+    let mut byte: u32;
+    let top_bit = 1u32 << 31; //31 is 32bit - 1
+
     for i in 0..256 {
-        let mut value = i as u32;
+        if true == rfl {
+            byte = reflect_long_32(i);
+        } else {
+            byte = i;
+        }
+
+        // Shift the cuttent table value "i" to the top byte in the long
+        let mut value: u32 = byte << 24; //24=32 bit - 8
+
+        // Step through all the bits in the byte
         for _ in 0..8 {
-            value = if (value & 1) == 1 {
-                (value >> 1) ^ poly
+            if (value & top_bit) != 0 {
+                value = (value << 1) ^ poly
             } else {
-                value >> 1
+                value <<= 1
             }
         }
-        table[i] = value;
+
+        if true == rfl {
+            value = reflect_long(value);
+        }
+
+        table[i as usize] = value;
     }
     table
 }
 
-pub fn make_table_crc64(poly: u64) -> [u64; 256] {
+/// Builds a CRC64 table using the standard CRC method
+/// If reflect==true, flip the individual byte bitwise, then flip the 32bit table value bitwise
+pub fn make_table_crc64(poly: u64, rfl: bool) -> [u64; 256] {
     let mut table = [0u64; 256];
+    let mut byte: u64;
+    let top_bit = 1u64 << 63; //63 is 64bit - 1
+
     for i in 0..256 {
-        let mut value = i as u64;
+        if true == rfl {
+            byte = reflect_byte_64(i);
+        } else {
+            byte = i as u64;
+        }
+
+        // Shift the cuttent table value "i" to the top byte in the long
+        let mut value: u64 = byte << 56; //56=64 bit - 8
+
+        // Step through all the bits in the byte
         for _ in 0..8 {
-            value = if (value & 1) == 1 {
-                (value >> 1) ^ poly
+            if (value & top_bit) != 0 {
+                value = (value << 1) ^ poly
             } else {
-                value >> 1
+                value <<= 1
             }
         }
-        table[i] = value;
+
+        if true == rfl {
+            value = reflect_long_long(value);
+        }
+
+        table[i as usize] = value;
     }
     table
 }
+
+/// Reflects a value of a 16 bit number
+pub fn reflect_short(mut value: u16) -> u16 {
```

Comment:
`reflect_16/32/64` or something similar?
just to make things consistent with next set of 3 functions